tsdb/store.go

```diff
@@ -164,13 +170,13 @@ func (s *Store) Close() error {
 	}
 	s.wg.Wait()
 
-	for _, sh := range s.shards {
-		if err := sh.Close(); err != nil {
+	for _, sl := range s.shardLocations {
+		if err := sl.Shard.Close(); err != nil {
```

Comment:
Possibly check for `nil`? Although `NewShard()` currently never returns nil, and that appears to be the only value that gets stored in `sl.Shard`, I see that you've checked for `nil` in other places for this field.